instructor/common.py

```diff
@@ -97,66 +97,71 @@ def assignment_setup(request):
                                   'error': error,
                                   'assignment_name': assignment_name,
                                   'based_on': request.session['based_on'],
-                                  'new': request.session['new']
+                                  'new': request.session['new'],
                               },
                               context_instance=RequestContext(request))
 
 @login_required
 def course_setup(request):
-    CourseForm = modelform_factory(models.Course, fields=['code', 'name'])
 
-    if request.method == "POST":
-        if 'course_pk' in request.POST:
-            # Adding to an existing course
-            course = models.Course.objects.get(pk=request.POST['course_pk'])
-        else:
-            # Create a new course
+    CourseFormSet = modelformset_factory(models.Course, extra=1, can_delete=True, fields=['name', 'code'])
+    course_selected = 0
+    if request.method == 'POST':
+
+        # change this course's name or code
+        formset = CourseFormSet(request.POST)
+        if formset.is_valid():
+            courses = formset.save(commit=False)
+            for obj in formset.deleted_objects:
+                obj.delete()
             user = User.objects.get(username=request.user)
-            form = CourseForm(request.POST)
-            if form.is_valid():
-                course = form.save(commit=False)
-                course.owner_id = user.id
+            for course in courses:
+                # need to set the owner
+                course.owner = user
```

Comment:
The data in `request.POST` could be anything so you can't trust that it only contains the user's course information. Can you add a check that says, if course.owner exists and is some other user, raise `django.core.exceptions.PermissionDenied`? This would prevent users from modifying courses they don't own.

This check should go near the beginning of this function so it happens before anything else.